wp/lib/bap_wp/src/precondition.ml

```diff
@@ -901,8 +901,12 @@ let visit_graph (env : Env.t) (post : Constr.t)
         let dst = G.Edge.dst e in
         debug "Entering back edge from\n%sto\n%s\n%!"
           (G.Node.to_string src) (G.Node.to_string dst);
-        let handler = Env.get_loop_handler env in
-        post, handler env post ~start:dst g
+        let tid = dst |> G.Node.label |> Term.tid in
```

Comment:
I'm wondering, is the general approach of how we handle this documented anywhere?

I guess this is more of a new issue than anything else.

Reply:
I could either put this in the comments for the loop invariant checker or in the design decisions markdown file. Leaning towards the design decisions.